### PR TITLE
Changes to Javascript code style

### PR DESCRIPTION
**Pull request by kolipka**
+ ESLint tool for verification

Google standard extended to AirBnB styleguide with one exception about line length.
It has a lot more rules, is more restrictive so unify code way better. Makes code  easier to read and prevent bugs.

Next step is to fix detected bugs and connect style check to travis-ci

IoT.js-DCO-1.0-Signed-off-by: Konrad Lipner k.lipner@samsung.com

### CODE REVIEW

**Comment by yichoi** [comment]
please clarify and specify the rules not change the rules.

File: docs/help/Coding-Style-Guideline.md [hunk 1]
```diff
@@ -160,47 +157,20 @@ Use either // or /* */ style comments. However, // style is much prefered.
 
 # Coding Style Guideline for Javascript
 
-This coding standard is based on [google javascript coding standard](https://google.github.io/styleguide/javascriptguide.xml)
+The Javascript code follows [AirBnB guideline](https://github.com/airbnb/javascript)
```
Comment (yichoi) [inline docs/help/Coding-Style-Guideline.md]: please clarify and specify the rules not change the rules.

Comment (kolipka) [inline docs/help/Coding-Style-Guideline.md]: All the rules are listed in attached link. With examples, rationale behind it, in many languages. I will not make it better, and copy/pasting it here will just add maintenace cost. I explained more here:
https://github.com/Samsung/iotjs/issues/631#issuecomment-281317920

Comment (nova0821) [inline docs/help/Coding-Style-Guideline.md]: @kolipka It looks URL(https://github.com/airbnb/javascript) which you link has rules for ES6.

Comment (kolipka) [inline docs/help/Coding-Style-Guideline.md]: You are right, however first link in this document points to ES5 rules extract. I will update my commit.

Comment (kolipka) [inline docs/help/Coding-Style-Guideline.md]: OK, I extended description with links to ESLint documentation and rules switches. Please re-check is it sufficient

**Comment by nova0821** [comment]
@kolipka I tried to run ``eslint`` with ``--fix`` option, it still has 2409 problems. Also, it generate new errors like line line length is over than 80 characters. I guess we need to fix them manually if we follow this eslint config, right?

Also, we have many codes from ``node.js`` and it looks them also generate errors with this eslint config. Do we need to fix codes from ``node.js``?

I'm worried that it will be huge task and also there will be mistakes if we need to fix them manually.
How do you think?

**Comment by kolipka** [comment]
@nova0821 If you approve this PR, I will do it by myself and put PR tomorrow EoD :D

To be serious - yes, there is a lot of simple errors, and few bigger. I will do it, but I suggest to put PR with separate commit per js file (or PR per file). It will be easy to review I hope.

**Comment by nova0821** [comment]
@kolipka Need an answer - We have many codes from ``node.js`` and it looks them also generate errors with this eslint config. Do we need to fix codes from ``node.js``? ``ble`` module also from node.js module - ``bleno``. We need to think about it seriously, because we need to maintenance them. It's hard to maintenance if we change them.

Also, we need to test all of them are working fine after you fix all errors from eslint. Are you sure it will be done in tomorrow? I think it's not simple task.

**Comment by kolipka** [comment]
@nova0821 In short: yes, we keep consistant style in our codebase. We exclude only deps from deps folder :)

When I put code in my repository I take full reponsibility about this code, despite it is taken from someone else. With moment it is included in my codebase, it is my code. I fix it to follow my quality standards, because not even single user cares that my product is broken because I copy/pasted low quality code from somewhere - it works or not. 
If I don't want this burden, I use it as a dependency. IMHO this is where bleno should end up (via npm). We don't like it, we contribute to their repository, or keep own fork. We don't care what is inside until it works as we require.

bleno use jshint - is it compatible with google guidelines? If no, we should leave it and disable quality checking? It is designed to work in node.js, not in Jerryscript - what we should do if it use too much memory or cause any problems?

Copy/paste 3 libraries all with differend style to single code base and it will be a horror to maintain anyway.

Are you sure ble works now? From errors I saw during linting, it works by accident if it works. We have no tests to prove it works or not now.

**Comment by yichoi** [comment]
I would like to discuss this issue after 1.0 release

**Pull request closed without merging**